dictionary-builder/src/main.rs

```diff
@@ -57,285 +60,199 @@ fn main() {
-                                    vm: self.vm.take(),
-                                    obs: self.obs.take(),
-                                };
-                                self.state = XmlReadingState::InTable;
-                                return Some(Ok(out));
-                            } else if local_name == b"tbody" {
-                                // the table ended!
-                                break;
-                            }
-                        }
-                    }
-                }
-                Ok(Event::Text(data)) => match self.state {
-                    XmlReadingState::InCellTag => {
-                        let data = data
-                            .unescape_and_decode(&self.parser)
-                            .unwrap()
-                            .replace("\u{200b}", "");
-                        self.tag = Some(data);
-                    }
-                    XmlReadingState::InCellName => {
-                        let data = data
-                            .unescape_and_decode(&self.parser)
-                            .unwrap()
-                            .replace("\u{200b}", "");
-                        self.name = Some(data);
-                    }
-                    XmlReadingState::InCellKeyword => {
-                        let data = data
-                            .unescape_and_decode(&self.parser)
-                            .unwrap()
-                            .replace("\u{200b}", "");
-                        self.keyword = Some(data);
-                    }
-                    XmlReadingState::InCellVR => {
-                        let data = data
-                            .unescape_and_decode(&self.parser)
-                            .unwrap()
-                            .replace("\u{200b}", "");
-                        self.vr = Some(data);
-                    }
-                    XmlReadingState::InCellVM => {
-                        let data = data
-                            .unescape_and_decode(&self.parser)
-                            .unwrap()
-                            .replace("\u{200b}", "");
-                        self.vm = Some(data);
-                    }
-                    XmlReadingState::InCellObs => {
-                        let data = data
-                            .unescape_and_decode(&self.parser)
-                            .unwrap()
-                            .replace("\u{200b}", "");
-                        self.obs = Some(data);
-                    }
-                    _ => {}
-                },
-                Ok(Event::Eof { .. }) => {
-                    break;
-                }
-                Ok(_) => {}
-                Err(e) => {
-                    return Some(Err(e));
-                }
-            }
-        }
+        writeln!(f, "/// {} {} {} {} {}", alias, tag, vr, vm, obs)?;
 
-        None
+        writeln!(
+            f,
+            "pub const {}: &str = \"{}\"; // {}",
```

Comment:
How does that play into the API? Would the ```DicomObject``` api stay the same? Accepting a string? Then, we'd use these const values like ``` obj.element_by_name(DICOM_FIELD.Key)?.to_str()?;```. Or, we would change ```element_by_name``` to accept a ```Tag```? What about unknown tags?

Reply:
Unless I'm missing something obvious, we'd just use the [`element`](https://docs.rs/dicom-object/0.1.0/dicom_object/trait.DicomObject.html#tymethod.element) method, which takes a tag. They could also sit in a module named `tags`, so it stays clear that the declarations are tags. In either case, if the tag is not known or is private, this would have to be defined by consumers or third parties anyway.

```rust
obj.element(tags::MODALITY)?.to_str()?;
```